### ISSUE TITLE
API & Query Explorer return different (larger) values when adding a Dimension

### ISSUE THREAD

**Issue by wInevitable** [googleanalytics/ga-dev-tools#355]
Hello,

We have been using the GA API recently for tracking inbound traffic sources and noticed the data (specifically goal1Completions) did not match our own records so I opened up the [Query Explorer](https://ga-dev-tools.appspot.com/query-explorer/) to take a closer look.

When I include a dimension (in this case ga:socialNetwork), the sum of the returned values for ga:users and ga:goal1Completions across each dimension (not set, Facebook, Instgram, etc) is greater than the values when I do not add a dimension. From what I've read, a dimension should only split the data across multiple subcategories not introduce new results.

I'm attaching two screenshots to show the issue more clearly. For example, in the images below you will see that when this particular query is broken down by the social network dimension there are 
```
(not set) - 135 users and 74 goal 1 completions
Facebook - 26 users and 0 goal 1 completions
Instagram - 3 users and 0 goal 1 competions
```
That's a total of 164 users and 74 goal 1 completions, but when I remove the dimension, we see only a total of 143 users and 54 goal 1 completions. Nothing else was changed in the query between execution calls. It's not clear how a single dimension, `(not set)`, could return more goal 1 Completions than when no dimension is set. Am I misunderstanding how dimensions work? I have also confirmed that the same data is returned when querying the API directly from our code. 

Thanks for your help!

![screen shot 2018-03-20 at 11 05 18 am](https://user-images.githubusercontent.com/6195283/37663576-def31532-2c2f-11e8-9605-acde33d6c966.png)
![screen shot 2018-03-20 at 11 04 40 am](https://user-images.githubusercontent.com/6195283/37663577-df045eaa-2c2f-11e8-8477-7bfd26b4c0c7.png)

**Comment by philipwalton** [comment]
> Am I misunderstanding how dimensions work?

I little bit. `ga:socialNetwork` is a session-level dimension that you're comparing to the user-level metric `ga:users`. Since users can have multiple sessions, it can be the case that the same user has a session where the social network value is "Facebook" and another session where the social network is "(not set)".

**Comment by wInevitable** [comment]
@philipwalton thanks for the fast response!

We're more concerned with the Goal 1 Completions than the Users. When we segment by any metric, city, social network, etc, we have seen the Goal 1 Completion value both above and below the value with no dimension added. I'm attaching screenshots from the latter case where the value is lower when dimensions are added.

We tried adding an audience segment where sessions includes Goal 1 Completion is greater than 1. We also tried another segment where users includes Goal 1 Completion is greater than 1. Both times the result was 0.

We've also noticed the date range having an effect. We don't see this error at smaller ranges (1 week for example). It's only when running longer queries (example 3 months).

In summary, we're confused how the Goal 1 Completion value could go either up or down for different users when we add a dimension. Could this be due to data sampling?

![screen shot 2018-03-23 at 10 43 30 am](https://user-images.githubusercontent.com/6195283/37836064-f9ab6b9a-2e87-11e8-88be-b8d8016ef25d.png)
![screen shot 2018-03-23 at 10 43 13 am](https://user-images.githubusercontent.com/6195283/37836066-fa0f29b4-2e87-11e8-96db-49b05587b47a.png)

**Comment by philipwalton** [comment]
Yes, sampling could absolutely have something to do this with. Also, adding/removing dimensions could as well, since some dimensions (e.g. `ga:eventLabel` or any custom dimension) don't have a default `(not set)` value.

If you're still concerned this is an error, I'd recommend filing a bug on the analytics issue tracker for the API, since you've said the results for the Query Explorer are the same:
https://issuetracker.google.com/issues?q=componentid:187400
